pkg/pillar/zedcloud/tls.go

```diff
@@ -21,8 +30,113 @@ const (
 	deviceCertName  = identityDirname + "/device.cert.pem"
 	deviceKeyName   = identityDirname + "/device.key.pem"
 	rootCertName    = identityDirname + "/root-certificate.pem"
+	tpmLockName     = "/var/tmp/zededa/tpm.lock"
 )
 
+//Custom crypto.PrivateKey interface implementation
+type TpmPrivateKey struct {
+	PublicKey crypto.PublicKey
+}
+
+//Helper structure to pack ecdsa signature for ASN1 encoding
+type ecdsaSignature struct {
+	R, S *big.Int
+}
+
+//Is the current device TPM enabled?
+func isTpmDevice() bool {
+	_, err := os.Stat(tpmmgr.TpmEnabledFile)
+	return (err == nil)
+}
+
+func (s TpmPrivateKey) Public() crypto.PublicKey {
+	clientCertName := "/config/device.cert.pem"
+	clientCertBytes, err := ioutil.ReadFile(clientCertName)
+	if err != nil {
+		return nil
+	}
+	block, clientCertBytes := pem.Decode(clientCertBytes)
+	var cert *x509.Certificate
+	cert, _ = x509.ParseCertificate(block.Bytes)
+	ecdsaPublicKey := cert.PublicKey.(*ecdsa.PublicKey)
+	return ecdsaPublicKey
+}
+
+func lockTpmAccess() {
+	for os.Mkdir(tpmLockName, 0750) != nil {
+		log.Debugln("Waiting for TPM lock.")
+		time.Sleep(1000 * time.Millisecond)
+	}
+	//XXX check if this sleep is still required
+	time.Sleep(1000 * time.Millisecond)
+}
+
+func unlockTpmAccess() {
+	os.Remove(tpmLockName)
+}
+
+func Sign(digest []byte) (*big.Int, *big.Int, error) {
```

Comment:
Makes sense to name this function tpmSign()

I think you can put the new functions in a new tpm.go file.

Reply:
Addressed in the lastest commit.